### PR TITLE
Core: Warn about deprecated config files

### PR DESCRIPTION
**Pull request by yannbf**
Issue: There's no issue specifically, but it's to avoid having more issues like #9854 

## What I did
This PR throws an error if the user does not have presets.js or main.js in their project. 
It could happen either if the file is really not there or if the file is being used in an unsupported extension. 
This could help users migrate by giving them a direction when upgrading storybook.

## How to test
- Does this need an update to the documentation?
Yes. I'm pointing to 6.0 docs in the log, so the docs from next need to be merged to master.

### CODE REVIEW

**Review by shilman — APPROVED** [review]
LGTM! Great fix!

**Review by shilman — CHANGES_REQUESTED** [review]
What if the user is still using `config.js`?

**Comment by yannbf** [comment]
> What if the user is still using `config.js`?

You are right. The intention would be to make people move away to the new configuration setup but it's better to warn with a deprecated message instead. Changes on the way!

**Comment by yannbf** [comment]
Oops! sorry for inactivity, corona side effects 😓 Work will be done soon, just need to get some feedback from @shilman  👍

**Comment by yannbf** [comment]
Hey @shilman changes are added! Please check if they all make sense to you.
ps: this PR will have a really small conflict with  #10288

File: lib/core/src/server/utils/interpret-files.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import fs from 'fs';
 import { extensions } from 'interpret';
 
-const boost = new Set(['.js', '.jsx', '.ts', '.tsx']);
+export const boost = new Set(['.js', '.jsx', '.ts', '.tsx']);
```
Comment (yannbf) [inline lib/core/src/server/utils/interpret-files.js:4]: This is where the conflict with #10288 will happen

**Review by shilman — CHANGES_REQUESTED** [review]
Can you update `exists` to handle the extensions code, i.e. `exists('main')`? will make the code a lot cleaner.

Also the other PR has been merged, so you can resolve the conflicts while you're at it! 😘

**Comment by yannbf** [comment]
> Can you update `exists` to handle the extensions code, i.e. `exists('main')`? will make the code a lot cleaner.
Thanks for you review! Done!

> Also the other PR has been merged, so you can resolve the conflicts while you're at it! 😘
Also done!

**Review by shilman — APPROVED** [review]
Looking great!! 😍

**Review by ndelangen — APPROVED** [review]

**Pull request merged**